### ISSUE TITLE
Phi-2 support

### ISSUE THREAD

**Issue by Herz3h** [PromtEngineer/localGPT#692]
Llama.cpp recently added support for [Phi-2](https://huggingface.co/microsoft/phi-2) model (https://github.com/ggerganov/llama.cpp/pull/4490)

Since this is using llama.cpp, I've tried configuring the model like this:

```python 
MODEL_ID = "TheBloke/phi-2-GGUF"
MODEL_BASENAME = "phi-2.Q6_K.gguf"
```

But it didn't work, and got an error (probably due to not updated llama.cpp):
```
error loading model: unknown model architecture: 'phi2'
llama_load_model_from_file: failed to load model
```

How to make it work ?

**Comment by PromtEngineer** [comment]
Yes, we will need to update the llamacpp version. I am planning on testing with updated versions of most of the packages.
